GentApp/Services/BranchService.cs

```diff
@@ -33,26 +32,40 @@ public async Task<IEnumerable<Branch>> GetBranches()
 			return Enumerable.Empty<Branch>();
 		}
 
-		public async Task<IEnumerable<Branch>> GetBranchesOfCompany(string id) {
-			// TODO: aanpassen
-			HttpResponseMessage response = await HttpClient.GetAsync(apiUrl);
-			if ( response.IsSuccessStatusCode ) {
-				return JsonConvert.DeserializeObject<IEnumerable<Branch>>(await response.Content.ReadAsStringAsync());
-			}
-			return Enumerable.Empty<Branch>();
-		}
-
-		public async Task Save(Branch branch)
+		public async Task<Branch> Save(Branch branch)
 		{
 			try
 			{
-				var response = await HttpClient.PostAsync(apiUrl, new StringContent(JsonConvert.SerializeObject(branch), Encoding.UTF8, "application/json"));
-				Console.WriteLine("test");
+				var request = new HttpRequestMessage {
+					Method = HttpMethod.Post,
+					RequestUri = new Uri(apiUrl),
+					Content = new StringContent(JsonConvert.SerializeObject(branch), Encoding.UTF8, "application/json")
+				};
+
+				var response = await HttpClient.SendAsync(request);
+
+				return JsonConvert.DeserializeObject<Branch>(response.Content.ToString());
 			}
 			catch (Exception ex)
 			{
 				await new MessageDialog(ex.Message).ShowAsync();
 			}
+
+			return null;
+		}
+
+		public async Task Update(Branch branch) {
+			try {
```

Comment:
wrm method voor branch up te daten, doen we niet via updateCompany?

Reply:
Keb alles moeten aanpassen zodawe enkele requests gebruiken. Tis redelijk ingewikkeld om een parent object up te daten en dan alle children ook up te daten in 1 keer. Ik dacht dat da vanzelf ging maar blijkbaar niet :/

Reply:
ok